### ISSUE TITLE
bug: TypeError: dispatcher.useInsertionEffect is not a function

### ISSUE THREAD

**Issue by michaelmikh** [trpc/trpc#3535]
### Provide environment information

```
System:
    OS: Linux 5.15 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (2) x64 AMD EPYC
    Memory: 1.61 GB / 3.84 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
    Watchman: 20220528.183901.0 - /usr/local/bin/watchman
  npmPackages:
    @tanstack/react-query: ^4.3.8 => 4.20.4 
    @trpc/client: ^10.7.0 => 10.7.0 
    @trpc/next: ^10.7.0 => 10.7.0 
    @trpc/react-query: ^10.7.0 => 10.7.0 
    @trpc/server: ^10.7.0 => 10.7.0 
    next: 13.0.7 => 13.0.7 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: 4.9.4 => 4.9.4 
```

### Describe the bug

TLDR: once the working Next app is wrapped with `trpc.withTRPC(...)` (with `ssr: true` in trpc config) I get the `TypeError: dispatcher.useInsertionEffect is not a function`.

I'm trying to use Mantine UI library (100% compatible with Next/SSR). Plain Next application works just fine, but as soon as I add a minimal amount of tRPC stuff and try to use it with `ssr` option turned `on`, I get this error. I'm not really sure it has something to do with the UI library I use as plain Next application works fine with all its SSR stuff.

I do understand that I can turn off the `ssr` setting, but that is more of a workaround.

<details>
  <summary>Error screenshot</summary>
  
  
  <img width="1512" alt="Next tRPC typeerror" src="https://user-images.githubusercontent.com/20373014/210368385-666a6491-0bbf-4c09-9e54-98292edfe8dc.png">
  
</details>

Google/SO has mentioned this kind of error only for unmatched `react` and `react-dom` packages versions, but I'm not sure that's the case either.

### Link to reproduction

https://codesandbox.io/p/sandbox/next-trpc-useinsertionhook-error-g19c5t

### To reproduce

To get the error please turn the `ssr` setting to `true` in `utils/trpc.ts` and refresh the page

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by KATT** [comment]
Not a tRPC-issue, this is a problem with that library not working in SSR. Also, see #596
